units/src/amount.rs

```diff
@@ -1714,10 +1711,8 @@ mod tests {
         assert_eq!(f(42.123456781, D::Bitcoin), Err(ParseAmountError::TooPrecise));
         assert_eq!(sf(-184467440738.0, D::Bitcoin), Err(ParseAmountError::TooBig));
         assert_eq!(f(18446744073709551617.0, D::Satoshi), Err(ParseAmountError::TooBig));
-        assert_eq!(
-            f(SignedAmount::MAX.to_float_in(D::Satoshi) + 1.0, D::Satoshi),
-            Err(ParseAmountError::TooBig)
-        );
+        assert!(f(SignedAmount::MAX.to_float_in(D::Satoshi) + 1.0, D::Satoshi).is_ok());
```

Comment:
Oh, I was confused before thinking this test is testing that the amount above max in float won't get converted. :man_facepalming: But it looks like that's what the test should've been doing from the start and it should just call `sf`, not `f`.

Reply:
Added two new patches that improve this unit test.